src/main.cpp

```diff
@@ -1199,20 +1173,31 @@ bool AcceptCertificateToMemoryPool(CTxMemPool& pool, CValidationState &state, co
     }
 
     string reason; // Rather not work on nonstandard transactions (unless -testnet/-regtest)
-    if (getRequireStandard() && !cert.IsStandard(reason, nextBlockHeight))
+    if (getRequireStandard() &&  !IsStandardTx(cert, reason, nextBlockHeight))
         return state.DoS(0, error("AcceptCertificateToMemoryPool: nonstandard certificate: %s", reason),
                             REJECT_NONSTANDARD, reason);
 
     //ABENEGIA: if (!tx.CheckFinal(STANDARD_LOCKTIME_VERIFY_FLAGS))// as of now certificate finality has yet to be defined
 
-    // is it already in the memory pool?
     uint256 certHash = cert.GetHash();
-    if (pool.existsCert(certHash))
+
+    // Check if cert is already in mempool or if there are conflicts with in-memory certs
     {
-        LogPrint("mempool", "Dropping certificateId %s : already in mempool\n", certHash.ToString());
-        return false;
+        LOCK(pool.cs);
```

Comment:
Consider moving these checks in the block below, where the lock on mempool is taken again

Reply:
We already touched upon this point in the past. I agree that locking once may be more efficient (not sure how to measure it though). However the two locked blocks gives better readability since they correspond to two different operations: first block contains "light" checks which peek into mempool datastructs only, while second blocks contains "less-light" checks picking coins from the caches system. I'd keep them separated for readability, at least currently